lib/ts/recipe/session/utils.ts

```diff
@@ -90,3 +92,19 @@ export const getFailureRedirectionInfo = async ({
         failedClaim,
     };
 };
+
+export function compareRedirectionURLToCurrentURL(redirectURL: string): boolean {
+    const currentUrl = WindowHandlerReference.getReferenceOrThrow().windowHandler.location.getHref();
+    let fullRedirectURL;
+    try {
+        new URL(redirectURL);
+        // if the url is a full, valid url, we can use that
+        fullRedirectURL = redirectURL;
+    } catch {
+        const appInfo = SuperTokens.getInstanceOrThrow().appInfo;
+        // otherwise we prepend the websiteDomain
+        fullRedirectURL = `${appInfo.websiteDomain.getAsStringDangerous()}${redirectURL}`;
```

Comment:
why should normalise the redirecturl using normalise path? Cause if the dev gives something like `second-factor` instead of `/second-factor`, then the resulting fullRedirectUrl will always be bad, and never match the current url anyway. But im not sure if normalising will drop query params / fragment etc. 

Also, maybe it's worth checking if the resulting fullRedirectUrl is a valid url and if it's not, then what do we do? Maybe we should throw an error in that case cause it's a bad developer config.

Reply:
We are not normalising the redirectUrl because we are not  normalizing it when we are "executing" the redirection either. If the dev gives `second-factor` (let's say on `localhost:3000/dashboard`), then the redirect will go to `localhost:3000/dashboard/second-factor`. 
This makes sense: the dev gave a relative redirection url, which should always redirect you. This means that the behaviour is as expected (it'll not match the current url, the helper returns false). I can add this as an explicit case if you prefer.